src/main/java/org/fenixedu/academic/service/services/enrollment/shift/WriteStudentAttendingCourse.java

```diff
@@ -40,7 +40,16 @@ protected void run(Registration registration, String executionCourseId) throws F
         if (registration == null) {
             throw new FenixServiceException("error.registration.not.exist");
         }
-        registration.addAttendsTo(readExecutionCourse(executionCourseId));
+
+        ExecutionCourse executionCourse = readExecutionCourse(executionCourseId);
+
+        for (Registration registration2 : registration.getStudent().getAllRegistrations()) {
+            if (!registration.getExternalId().equals(registration2.getExternalId()) && registration2.attends(executionCourse)) {
```

Comment:
The fenix-framework guarantees that each instance of a domain object is unique within the virtual machine. Therefore no need exists to compare the objects external id. Just compare the objects instead. The resulting code is easier to read and more efficient.